core/shared/src/main/scala/scodec/bits/BitVector.scala

```diff
@@ -1320,6 +1323,30 @@ sealed abstract class BitVector extends BitwiseOperations[BitVector, Long] with
     )
 
   protected final def writeReplace(): AnyRef = new SerializationProxy(toByteArray, size)
+
+  override def compare(that: BitVector): Int =
+    if (this.eq(that)) {
+      0
+    } else {
+      val thisLength = this.length
+      val thatLength = that.length
+      val commonLength = thisLength.min(thatLength)
+      var i = 0
+      while (i < commonLength) {
+        val cmp = this(i).compare(that(i))
+        if (cmp != 0) {
+          return cmp
+        }
+        i = i + 1
+      }
+      if (thisLength < thatLength) {
```

Comment:
Why prefer comparing elements over length? As opposed to first comparing length and only if equal, comparing elements?

Reply:
This is supposed to do a [_lexicographical_ ordering](https://en.wikipedia.org/wiki/Lexicographical_order) of the Byte vectors. Lengths are compared _only after_ we have established that one is a prefix of the other.
Before you do that (i.e. you compare their elements), the lengths alone can't tell you which ByteVector is greater.
See for example this test case:
```scala
assert(ByteVector(1, 2) < ByteVector(2))
```
The first one may be longer, but is still lesser that the second one.